shared-bindings/tilepalettemapper/TilePaletteMapper.c

```diff
@@ -0,0 +1,165 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Tim Cocks for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+#include <stdint.h>
+#include <stdlib.h>
+#include "shared/runtime/context_manager_helpers.h"
+#include "py/binary.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "shared-bindings/util.h"
+#include "shared-bindings/displayio/Palette.h"
+#include "shared-bindings/tilepalettemapper/TilePaletteMapper.h"
+
+//| class TilePaletteMapper:
+//|     """Contains a displayio.Palette a series of mappings from one palette color_index to another palette color_index.
+//|     One mapping for tile in a 2D grid. Can be set as the pixel_shader for a TileGrid to re-map the colors for
+//|     specific tile(s) within the TileGrid."""
+//|
+//|     def __init__(self, palette: displayio.Palette, width: int, height: int) -> None:
+//|         """Create a TilePaletteMApper object to store a set of color mappings for tiles.
+//|
+//|         :param displayio.Palette palette: The palette to get mapped colors from.
```

Comment:
We may want to expand this later to take ColorConverter. That way we'd be able to do full color text rendering.

Reply:
Would it have some other mechanism for mapping the colors? The way it is now I think this would require one array of length equal the number of colors in the colorspace for each tile. Is there enough room to store that?

Reply:
You'd store one 24 bit color for every input value. So if it is 1 bit color, you store 2 RGB colors that then are converted via ColorConverter. This is how non-indexed bitmaps, OnDiskBitmaps works. The value *is* the color in that case. ColorConverter just switches it to the output color space.